src/WebJobs.Script/FileAugmentation/PowerShell/PowerShellFileAugmentor.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.FileAugmentation.PowerShell
+{
+    internal class PowerShellFileAugmentor : IFuncAppFileAugmentor
+    {
+        /// <summary>
+        /// Adds the required files to the function app
+        /// </summary>
+        /// <param name="scriptRootPath">The root path of the function app</param>
+        /// <returns>An empty completed <see cref="Task"/> object with</returns>
+        public Task AugmentFiles(string scriptRootPath)
+        {
+            if (string.IsNullOrWhiteSpace(scriptRootPath))
+            {
+                throw new ArgumentException("The parameter {0} cannot be null or empty", nameof(scriptRootPath));
+            }
+
+            AddOrUpdateHostFile(scriptRootPath);
+            AddRequirementsFile(scriptRootPath);
+            AddProfileFile(scriptRootPath);
+            return Task.CompletedTask;
+        }
+
+        private void AddOrUpdateHostFile(string scriptRootPath)
+        {
+            string hostFilePath = Path.Combine(scriptRootPath, "host.json");
+            string hostJsonContent = null;
+            if (File.Exists(hostFilePath))
+            {
+                using (StreamReader r = new StreamReader(hostFilePath))
+                {
+                    hostJsonContent = r.ReadToEnd();
+                }
+            }
+
+            if (string.IsNullOrWhiteSpace(hostJsonContent))
+            {
+                hostJsonContent = "{'version': '2.0'}";
+            }
+
+            var hostJsonJObj = JObject.Parse(hostJsonContent);
+            var managedDependency = hostJsonJObj.GetValue("managedDependency");
+            if (managedDependency == null)
```

Comment:
Will this become a requirement in the PowerShell scenarios? If customers remove the `manageDependency` property, this will always re-add that on startup.

Reply:
I'll discuss about this scenario with Eamon and update accordingly.